File: HomeWork_3/src/com/valery/homework3/PartOne/Book.java
package com.valery.homework3.PartOne;

public class Book implements Publication {
    private final double price;
    private final int wordsCount;

    public Book(double price, int wordsCount) {
        this.price = price;
        this.wordsCount = wordsCount;
    }

    @Override
    public double getPrice() {
        return this.price;
    }

    @Override
    public String getType() {
        if (wordsCount > 7500) {
            return "Novel";
        } else if (wordsCount > 1000) {
            return "Short Story";
        } else {
            return "Flash Fiction";
        }
    }

    @Override
    public int getWordsCount() {
        return wordsCount;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Book book)) return false;
        return Double.compare(book.price, price) == 0 && wordsCount == book.wordsCount;
    }
}

File: HomeWork_1/src/main/kotlin/Book.kt
class Book(override val price: Double, override val wordCount: Int): Publication {
    override fun getType(): String = when (wordCount) {
        in 0..1000 -> "Flash Fiction"
        in 1001..7500 -> "Short Story"
        else -> "Novel"
    }

    override fun equals(other: Any?): Boolean {
        val isEqualByReference = this === other
        val isEqualByValue = other is Book && this.price == other.price && this.wordCount == other.wordCount
        println("Comparing books by reference: $isEqualByReference")
        println("Comparing books by value: $isEqualByValue\n")
        return isEqualByValue
    }
}